problem_builder/tests/unit/test_instructor_tool.py

```diff
@@ -0,0 +1,226 @@
+            },
+        ]
+
+    def test_student_view_block_tree_names(self):
+        """
+        Check if `student_view` method uses preferred block attributes
+        when building block tree for render context.
+
+        Each entry of a block tree is a dictionary that must contain
+        an `id` key and a `name` key.
+
+        - `id` must be set to the ID
+          (usage_id or block_id) of the corresponding block.
+
+        - `name` must be set to the value of one of the following
+          attributes of the corresponding block:
+
+          - question
+          - name (question ID)
+          - display_name (question title)
+          - block ID
+
+          Note that the attributes are listed in order of preference;
+          i.e., if `block.question` has a meaningful value, that value
+          should be used for `name` (irrespective of what the values
+          of the other attributes might be).
+        """
+        instructor_tool.loader = Mock()
+        instructor_tool.loader.render_template.return_value = u''
+
+        self.block.student_view(context={})
+        block_tree = instructor_tool.loader.render_template.call_args[0][1]['block_tree']
+
+        # Check size of block_tree
+        num_blocks = len(self.block.children) + 1
+        self.assertEqual(len(block_tree), num_blocks)
+
+        # Check block_tree for root entry
+        # - Does block_tree contain single entry whose `id` matches usage_id of instructor block?
+        matching_blocks = [block for block in block_tree if block['id'] == self.block.scope_ids.usage_id]
+        self.assertEqual(len(matching_blocks), 1)
+        # # - Is `name` of that entry set to "All"?
+        matching_block = matching_blocks.pop()
+        self.assertEqual(matching_block['name'], 'All')
+
+        # Check block_tree for children
+        for child in self.block.children:
+            usage_id = child.get('usage_id')
+            attrs = child.get('attrs', {})
+            # - Does block_tree contain single entry whose `id` matches usage_id of current child?
+            matching_blocks = [block for block in block_tree if block['id'] == usage_id]
+            self.assertEqual(len(matching_blocks), 1)
+            # - Is `name` of that entry set to value of current child's preferred_attr?
+            matching_block = matching_blocks.pop()
+            if not attrs:
+                self.assertEqual(matching_block['name'], usage_id)
+            else:
+                preferred_attr = child.get('preferred_attr')
+                self.assertEqual(matching_block['name'], attrs[preferred_attr])
+
+    def test_student_view_block_tree_choice_blocks_excluded(self):
+        """
+        Check if `student_view` method excludes 'pb-choice' blocks when
+        building block tree for render context.
+        """
+        # Pretend that all blocks in self.block.children are of type 'pb-choice:
+        self.block.runtime.id_reader = Mock()
```

Comment:
@itsjeyd nit: I would access runtime mock using `self.runtime_mock` to make clear that it's actually a mock.